crates/router/src/connector/bambora/transformers.rs

```diff
@@ -45,16 +50,72 @@ pub struct Browser {
 pub struct BamboraPaymentsRequest {
     amount: i64,
     payment_method: PaymentMethod,
+    customer_ip: Option<std::net::IpAddr>,
+    term_url: Option<String>,
     card: BamboraCard,
 }
 
+#[derive(Debug, Serialize, Deserialize)]
+pub struct BamboraRedirectionResponse {
+    pub cres: String,
+}
+
+fn get_browser_info(item: &types::PaymentsAuthorizeRouterData) -> Option<BamboraBrowserInfo> {
+    if matches!(item.auth_type, enums::AuthenticationType::ThreeDs) {
+        item.request
+            .browser_info
+            .as_ref()
+            .map(|info| BamboraBrowserInfo {
+                accept_header: info.accept_header.clone(),
+                java_enabled: info.java_enabled,
+                language: info.language.clone(),
+                color_depth: info.color_depth,
+                screen_height: info.screen_height,
+                screen_width: info.screen_width,
+                time_zone: info.time_zone,
+                user_agent: info.user_agent.clone(),
+                javascript_enabled: info.java_script_enabled,
+            })
+    } else {
+        None
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct CresBambora {
```

Comment:
Is Serialize required for this struct?

For response struct's Serialize not required.